feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -119,4 +119,14 @@ export class ConfiguratorAttributeBaseComponent {
     }
     return attributeUiKey;
   }
+
+  /**
+   * Creates a unique key for focus handling for the given attribute and value
+   * @param attributeId
+   * @param valueCode
+   * @returns focus key
+   */
+  createFocusId(attributeId: string, valueCode: string): string {
```

Comment:
I know we try to avoid breaking changes. But, can we make this one as service?

Reply:
Agree that a service would fit better here. But we should not do this for 3.3, as it would create breaking changes to nearly all of our attribute type components.
Create #12039 do be considered later (not necessarily in 4.0, let's see if me manage)